### ISSUE TITLE
Call to dcim.devices.all() fails with a 504 when using nginx and uwsgi

### ISSUE THREAD

**Issue by ozmusic** [nautobot/nautobot#1329]
### Environment
* Python version: 3.8.11
* Nautobot version: 1.2.3
* nginx version: 1:1.16-1-1.el7.ngx
* uwsgi version: 2.0.19.1

### Steps to Reproduce
1. Deploy nautobot with nginx and uwsgi
2. Import a number of devices into nautobot
3. Call dcim.devices.all (either through restapi or pynautobot)

### Expected Behaviour
All devices should be returned via the api
Note: this worked against demo.nautobot.com (~300 devices), local install has ~5200 devices

### Observed Behaviour
504 gateway timeout from the server

client log:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/demo/Library/Python/3.8/lib/python/site-packages/pynautobot/core/endpoint.py", line 99, in all
    return response_loader(req.get(), self.return_obj, self)
  File "/Users/demo/Library/Python/3.8/lib/python/site-packages/pynautobot/core/query.py", line 295, in get
    return req_all()
  File "/Users/demo/Library/Python/3.8/lib/python/site-packages/pynautobot/core/query.py", line 253, in req_all
    req = self._make_call(add_params=add_params)
  File "/Users/demo/Library/Python/3.8/lib/python/site-packages/pynautobot/core/query.py", line 226, in _make_call
    raise RequestError(req)
pynautobot.core.query.RequestError: The request failed with code 504 Gateway Time-out but more specific details were not returned in json. Check the Nautobot Logs or investigate this exception's error attribute.
```

uwsgi log:
```
Thu Feb  3 11:35:17 2022 - uwsgi_response_writev_headers_and_body_do(): Broken pipe [core/writer.c line 306] during GET /api/dcim/devices/ (192.0.2.55)
OSError: write error
```

nginx log:
```
2022/02/03 11:34:56 [error] 28892#28892: *35577 upstream timed out (110: Connection timed out) while reading response header from upstream, client: 192.0.2.55, server: nautobot-dev-1.example.com, request: "GET /api/dcim/devices/ HTTP/1.1", upstream: "uwsgi://unix:///run/nautobot-dev/uwsgi.socket", host: "nautobot-dev-1.example.com"
```

**Comment by jathanism** [comment]
Can you please add a `?limit=100` or something to limit the number of results in the query to see if that comes back okay? I think we may have introduced a regression in v1.2.4 and would like to confirm if paginating the results still works. Thanks!

**Comment by ozmusic** [comment]
Limit 100 (500, and 1000) work OK for me, pagination in our config file is set to 1000.

BTW, this isn't happening for prefixes or IP addresses

**Comment by jathanism** [comment]
Device objects can end up quite large in the API. Can you please confirm that the results are being paginated?

Short of that, there might be a mismatch between server and Nginx config that could be tweaked? @nniehoff might have some thoughts?

**Comment by ozmusic** [comment]
I'm not seeing any of the signs of paginated requests in the nginx access logs
```192.0.2.55 - - [03/Feb/2022:17:46:52 +0000] "GET /api/dcim/devices/ HTTP/1.1" 499 0 "-" "python-requests/2.24.0"
192.0.2.55 - - [03/Feb/2022:17:47:54 +0000] "GET /api/dcim/devices/ HTTP/1.1" 504 167 "-" "python-requests/2.24.0"
192.0.2.55 - - [03/Feb/2022:17:48:32 +0000] "GET /api/dcim/interfaces/ HTTP/2.0" 504 167 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:96.0) Gecko/20100101 Firefox/96.0"
192.0.2.55 - - [03/Feb/2022:17:49:14 +0000] "GET /api/ipam/prefixes/ HTTP/2.0" 200 402963 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:96.0) Gecko/20100101 Firefox/96.0"
192.0.2.55 - - [03/Feb/2022:17:50:02 +0000] "GET /api/ipam/prefixes/ HTTP/1.1" 200 235553 "-" "python-requests/2.24.0"
```

**Comment by glennmatthews** [comment]
Current working theory is that this may be a breakage to pagination possibly introduced by #1079.

**Comment by nniehoff** [comment]
> Short of that, there might be a mismatch between server and Nginx config that could be tweaked? @nniehoff might have some thoughts?

@ozmusic I'm assuming your nginx.conf looks something like the documented config:

```no-highlight
server {
...
    location / {
        include uwsgi_params;
        uwsgi_pass  127.0.0.1:8001;
        uwsgi_param Host $host;
        uwsgi_param X-Real-IP $remote_addr;
        uwsgi_param X-Forwarded-For $proxy_add_x_forwarded_for;
        uwsgi_param X-Forwarded-Proto $http_x_forwarded_proto;
    }
}
```

Can you try adding `uwsgi_read_timeout 300s;` to the list of uwsgi_params?  This might just be a workaround for the issue but I'm curious if it helps.

**Comment by ozmusic** [comment]
@nniehoff another member in my team tried changing the timeout, this worked (with a response time of ~180 seconds)

**Comment by glennmatthews** [comment]
The underlying issue that brought this to light (a regression in our default pagination behavior) is fixed in #1332. It sounds like there's a workaround in the meantime of either increasing the uswgi timeout or explicitly setting a pagination `limit` value. @ozmusic are you good with closing this issue at this time or is there anything further here that needs pursuing?

**Comment by ozmusic** [comment]
@glennmatthews sure, I'm happy to close this out
